libsolidity/ast/CallGraph.h

```diff
@@ -62,7 +62,7 @@ struct CallGraph
 	std::map<Node, std::set<Node, CompareByID>, CompareByID> edges;
 
 	/// Contracts that may get created with `new` by functions present in the graph.
-	std::set<ContractDefinition const*, ASTNode::CompareByID> createdContracts;
+	std::map<ContractDefinition const*, ASTNode const*, ASTNode::CompareByID> createdContracts;
```

Comment:
I'd make it more specific and use `NewExpression` unless we know that it can indeed be an arbitrary node.

Also, there can be multiple places where a contract is created.

```suggestion
	std::map<ContractDefinition const*, std::set<NewExpression const*>, ASTNode::CompareByID> createdContracts;
```